### DISCUSSION TITLE
Use of custom identities

### DISCUSSION THREAD

**Discussion by eliasboegel** [sympy/sympy#26030] [General]
Hi all,

I work with Hermite functions and am just beginning to use sympy. I would like to help sympy with some identities that apply to Hermite functions:

1) A recursion relation (I would like to help sympy resolve $x^k\psi_n(x)$ terms into a sum of $c\psi_i(x)$ terms together with use of another relation for the derivative which I could implement with `fdiff()`):

$\sqrt{2(n+1)}\psi_{n+1}(x) = (x - \frac{d}{dx})\psi_n(x)$

2) An orthogonality relation

$\int_{-\infty}^\infty \psi_n(x) \psi_m(x) dx = \delta_{nm}$

I've defined the Hermite functions as a custom function (via deriving the Function class) with the built-in Hermite polynomials. How can I implement these identities? I would really like to avoid letting sympy compute everything from scratch.

Thanks!

**Comment by oscarbenjamin** [comment]
SymPy already has Hermite functions:
```
In [16]: hermite(n, x).diff(x)
Out[16]: 2⋅n⋅hermite(n - 1, x)
```
It is not clear to me what it is that you want to do or if you are proposing a change to SymPy.

If you are not proposing a change to SymPy then this issue should be closed because GitHub issues are not for asking help with using SymPy.

**Comment by eliasboegel** [comment]
These are Hermite polynomials, which I also use in my SymPy definition of the Hermite functions. Hermite functions are defined as such:

$\psi_n(x) = (2^nn! \sqrt{\pi}) ^{-\frac{1}{2} } e^{-\frac{x^2}{2} } H_n(x)$

with $H_n(x)$ being the Hermite polynomial(s). The Hermite functions come with additional identities such as the orthogonality relation in my original post, which I wanted to implement.

I suppose you could see this as purely a usage question of "how does one implement such identities and relations" if there is no interest in having Hermite functions as part of SymPy. Could you point me to the correct place to ask this question?

**Comment by oscarbenjamin** [comment]
If the Hermite functions are easily defined in terms of the Hermite polynomials then maybe it does not make sense to add them to SymPy. I don't see any examples of comparable or related systems (Mathematica, SciPy etc.) defining Hermite functions.

> Could you point me to the correct place to ask this question?

I will convert this issue into a "discussion" but I suggest that it might be better to ask on the mailing list instead.

**Reply by eliasboegel** [reply]
Thanks for converting to a discussion and thanks for the reference to the mailing list.

Yes, I see it the same way. I would rather just keep them out of SymPy. My particular issue is not per se that I can't represent them or compute with them, but rather that I need to evaluate the recursion relation and orthogonality relation a large number of times. So, I would love to make this more obvious for SymPy to work with by directly specifying these two relations without actually having to symbolically compute them (similar to how `hermite(n, x).diff(x)` is also directly defined and not computed and subsequently rewritten back as a Hermite polynomial).
